File: GroceryListManagerTest.py
import unittest

class GroceryListManager:
    def __init__(self):
        self.items = []

    def addItem(self, item):
        self.items.append(item)

    def removeItem(self, index):
        if index >= 0 and index < len(self.items):
            del self.items[index]

    def getItemsList(self):
        return self.items

class TestGroceryListManager(unittest.TestCase):

    def testAddItem(self):
        manager = GroceryListManager()
        manager.addItem("rice")
        manager.addItem("millet")
        items = manager.getItemsList()
        self.assertEqual(2, len(items))
        self.assertIn("rice", items)
        self.assertIn("millet", items)

    def testRemoveItem(self):
        manager = GroceryListManager()
        manager.addItem("rice")
        manager.addItem("millet")
        manager.removeItem(0)
        items = manager.getItemsList()
        self.assertEqual(1, len(items))
        self.assertNotIn("rice", items)
        self.assertIn("millet", items)

    def testShowItemsFromEmptyList(self):
        manager = GroceryListManager()
        items = manager.getItemsList()
        self.assertEqual(0, len(items))

    def testShowItemsWithStuffs(self):
        manager = GroceryListManager()
        manager.addItem("rice")
        manager.addItem("millet")
        items = manager.getItemsList()
        self.assertEqual(2, len(items))
        self.assertEqual("rice", items[0])
        self.assertEqual("millet", items[1])

File: DAY4/GroceryListManagerTest.java
import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;
import java.util.ArrayList;

public class GroceryListManagerTest {
    @Test
    public void testAddItem() {
        GroceryListManager manager = new GroceryListManager();
       
 	// Adding items
        manager.addItem("rice");
        manager.addItem("millet");

        // Checking items that were added
        ArrayList<String> items = manager.getItemsList();
        assertEquals(2, items.size());
        assertTrue(items.contains("rice"));
        assertTrue(items.contains("millet"));
    }

    @Test
    public void testRemoveItem() {
        GroceryListManager manager = new GroceryListManager();
        
	// Adding items
        manager.addItem("rice");
        manager.addItem("millet");

        // Remove an item (index 0 remove first item)
        manager.removeItem(0);

        // Check item that is removed
        ArrayList<String> items = manager.getItemsList();
        assertEquals(1, items.size());
        assertFalse(items.contains("rice"));
        assertTrue(items.contains("millet"));
    }

    @Test
    public void testShowItemsFromEmptyList() {
        GroceryListManager manager = new GroceryListManager();

        // Checking for empty list
        ArrayList<String> items = manager.getItemsList();
        assertEquals(0, items.size());
    }

    @Test
    public void testShowItemsWithStuffs() {
        GroceryListManager manager = new GroceryListManager();

        // Adding items
        manager.addItem("rice");
        manager.addItem("millet");

        // Check items that are present
        ArrayList<String> items = manager.getItemsList();

        // List should contain 2 items
        assertEquals(2, items.size());

        // First item should be rice
        assertEquals("rice", items.get(0));

        // Second item should be millet
        assertEquals("millet", items.get(1));
    }
}


public class GroceryListManager {
   private ArrayList<String> items;

public GroceryListManager () {
   manager.addItem = new ArrayList<>();
}

public void addItem(String item){
   manager.add(item);
}

public void removeItem(int index) {
if(index > 0 && index <= items.size()) {
	manager.removeItem(index);
}
}

public ArrayList<String> getItemsList() {
	return items;
}
}



